### PR TITLE
PotStageHistory module

### PR DESCRIPTION
**Pull request by mharacewiat**

### CODE REVIEW

File: src/Api/Modules/PotStageHistory.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    protected static $primary_key = 'POTENTIALSTAGEHISTORYID';
+
+    protected static $associated_entity = \Zoho\CRM\Entities\PotStageHistory::class;
+
+    protected static $supported_methods = [
+        'getRelatedRecords',
+    ];
+
+    public function getPotentialHistory($potentialId)
```
Comment (tristanjahier) [inline src/Api/Modules/PotStageHistory.php]: Variable naming: you must replace `$potentialId` with `$potential_id`.
This naming convention could change in a future release, but at least for now we shall stick to the same thing everywhere.

File: src/Api/Modules/PotStageHistory.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    {
+        return $this->getRelatedById('Potential', $potentialId)->fetch()->getContent();
```
Comment (tristanjahier) [inline src/Api/Modules/PotStageHistory.php]: The `->fetch()->getContent()` part must be called explicitely by the developer or is done automatically by the `Client`, therefore you must remove it.

To achieve the same thing automatically, you need to set these preferences:
1. `fetch()` is done by setting `'auto_fetch_paginated_requests'` to `true`.
2. `getContent()` is done by setting `'response_mode'` to `Zoho\CRM\ClientResponseMode::DIRECT`.

Notice the preference `'records_as_entities'`, used to transform records into entities automatically.

**Review by tristanjahier — CHANGES_REQUESTED** [review]
Nice work, I think you understood the structure of this library quite well, but I request 2 changes.

File: src/Api/Modules/PotStageHistory.php (hunk 2 above)
Comment (tristanjahier) [inline src/Api/Modules/PotStageHistory.php]: You are setting the `parentModule` (1st argument) of the request to `'Potential'`. But the documentation says it must be `'Potentials'` (plural), so even if it works either way, you must change it.

https://www.zoho.com/crm/help/api/getrelatedrecords.html#To_fetch_the_Stage_History_related_to_a_Potential

**Review by tristanjahier — CHANGES_REQUESTED** [review]

**Comment by mharacewiat** [comment]
Changes applied, please review.

**Comment by tristanjahier** [comment]
All good! Bravo! 👏

**Comment by tristanjahier** [comment]
All good! Bravo! 👏

**Pull request merged**